core/src/main/java/com/google/errorprone/bugpatterns/StatementSwitchToExpressionSwitch.java

```diff
@@ -122,7 +134,7 @@ private static AnalysisResult analyzeSwitchTree(SwitchTree switchTree) {
 
       List<? extends StatementTree> statements = caseTree.getStatements();
       CaseFallThru caseFallThru = CaseFallThru.MAYBE_FALLS_THRU;
-      if (statements == null || statements.isEmpty()) {
+      if (statements.isEmpty()) {
```

Comment:
This reverts the change in #3645, because there are actually many more `CaseTree#getStatements()` calls in this class, and they _all_ assume a  non-`null` result. The `isRuleSwitch` check now guarantees that.